ext/re2/re2.cc

```diff
@@ -122,18 +122,34 @@ static void parse_re2_options(RE2::Options* re2_options, const VALUE options) {
   }
 }
 
-static void re2_matchdata_mark(re2_matchdata* self) {
+static void re2_matchdata_mark(void * data) {
+  re2_matchdata* self = (re2_matchdata*)data;
   rb_gc_mark(self->regexp);
   rb_gc_mark(self->text);
 }
 
-static void re2_matchdata_free(re2_matchdata* self) {
+static void re2_matchdata_free(void * data) {
+  re2_matchdata* self = (re2_matchdata*)data;
   if (self->matches) {
     delete[] self->matches;
   }
   free(self);
```

Comment:
Should we be using `xfree` here (and all other dfree functions) as per @peterzhu2118’s https://blog.peterzhu.ca/ruby-c-ext-part-7/#dfree?

Reply:
Fixed in 0bb9b0de7c9def5b7b439f0981ba6dbeb199245f.